### PR TITLE
Fixed bug in FileChooser for output directories.

### PR DESCRIPTION
**Pull request by AlbertoCasasOrtiz**
Fixes issue #1378 

### Brief summary of changes

When opening FileChooser to select the Output Directory on Analysis Tools, it was not updating the title and the filters. Thus, it was showing the title and filters set the last time you opened the FileChooser otherwhere (This is the reason the title and the filters were changing over time, it depends on the value assigned last time you opened a FileChooser). This has been solved by:

 - Added function to allow set title of the window without having to add elements to the FileFilter.
 - Reset FileFilter when the FileChooser is opened.
 - Make sure that the FileChooser is in save mode.

### Testing I've completed

Locally tested on Ubuntu 22.04.

### CHANGELOG.md (choose one)

- Updated.

### CODE REVIEW

**Comment by aymanhab** [comment]
Thanks @AlbertoCasasOrtiz Please resolve conflicts, the change looks good. Thank you.

File: Gui/opensim/utils/src/org/opensim/swingui/FileTextFieldAndChooser.java [hunk 1]
```diff
@@ -84,6 +84,10 @@ public void setExtensionsAndDescription(String extensions, String description) {
    public void setFileFilter(FileFilter filter) {
       this.filter = filter;
    }
+	
+   public void setDialogTitle(String title) {
+	  this.title = title;   
+   }
```
Comment (aymanhab) [inline Gui/opensim/utils/src/org/opensim/swingui/FileTextFieldAndChooser.java:91]: Are there scenarios where title wouldn't be cleared from one invocation to the next?

File: Gui/opensim/utils/src/org/opensim/utils/FileUtils.java [hunk 2]
```diff
@@ -262,6 +262,8 @@ public String browseForFolder(Frame parent, String description, boolean allowCre
         dlog.setCurrentDirectory(new File(defaultDir));
         dlog.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
         
+		dlog.resetChoosableFileFilters();
+
```
Comment (aymanhab) [inline Gui/opensim/utils/src/org/opensim/utils/FileUtils.java:266]: Please fix indentation

File: Gui/opensim/utils/src/org/opensim/swingui/FileTextFieldAndChooser.java (hunk 1 above)
Comment (AlbertoCasasOrtiz) [reply Gui/opensim/utils/src/org/opensim/swingui/FileTextFieldAndChooser.java:91]: I have tested and it always sets the new title. I don't think there is any scenario where it couldn't be cleared.

**Review by aymanhab — APPROVED** [review]
Thanks @AlbertoCasasOrtiz

**Pull request merged**